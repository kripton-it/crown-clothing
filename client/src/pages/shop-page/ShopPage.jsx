import React, { lazy, Suspense } from "react";
import { Route } from "react-router-dom";
import Spinner from "../../components/spinner/spinner";

import { ShopPageContainer } from "./ShopPageStyles";

const CollectionsList = lazy(() =>
  import("../../components/collections-list/CollectionsListContainer")
);
const CollectionPage = lazy(() =>
  import("../collection-page/CollectionPageContainer")
);

const ShopPage = ({ match }) => {
  return (
    <ShopPageContainer>
      <Suspense fallback={<Spinner />}>
        <Route exact path={`${match.path}`} component={CollectionsList} />
        <Route
          path={`${match.path}/:collectionId`}
          component={CollectionPage}
        />
      </Suspense>
    </ShopPageContainer>
  );
};

export default ShopPage;
