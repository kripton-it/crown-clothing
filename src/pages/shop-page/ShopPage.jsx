import React from "react";
import { Route } from "react-router-dom";

import CollectionsList from "../../components/collections-list/CollectionsList";
import CollectionPage from "../collection-page/CollectionPage";

const ShopPage = ({ match }) => {
  return (
    <div className="shop-page">
      <Route exact path={`${match.path}`} component={CollectionsList} />
      <Route path={`${match.path}/:collectionId`} component={CollectionPage} />
    </div>
  );
};

export default ShopPage;
