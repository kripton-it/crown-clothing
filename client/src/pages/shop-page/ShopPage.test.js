import React from "react";
import { mount } from "enzyme";
// import { shallow } from "enzyme";
import { BrowserRouter } from 'react-router-dom';

import ShopPage from "./ShopPage";

it("expect to render ShopPage page", () => {
  const mockMatch = {
    path: ''
  };
  
  const mockProps = {
    match: mockMatch,
  };

  const wrapper = mount(
    <BrowserRouter>
      <ShopPage {...mockProps} />
    </BrowserRouter>
  );

  expect(wrapper).toMatchSnapshot();
});
