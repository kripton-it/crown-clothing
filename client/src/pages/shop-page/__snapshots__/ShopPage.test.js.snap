// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ShopPage page 1`] = `ReactWrapper {}`;
